pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -458,14 +458,94 @@ def ctrl_decomp_bisect(
 
 
 def decompose_mcx(control_wires, target_wire, work_wires):
-    """Decomposes the multi-controlled PauliX gate"""
-
+    """Decomposes the multi-controlled PauliX gate using decompositions from
+    `Barenco et al. (1995) <https://arxiv.org/abs/quant-ph/9503016>`_"""
     num_work_wires_needed = len(control_wires) - 2
 
+    if len(control_wires) == 1:
+        return [qml.CNOT(wires=control_wires + Wires(target_wire))]
+    if len(control_wires) == 2:
+        return qml.Toffoli.compute_decomposition(wires=control_wires + Wires(target_wire))
+
     if len(work_wires) >= num_work_wires_needed:
+        # Lemma 7.2
         return _decompose_mcx_with_many_workers(control_wires, target_wire, work_wires)
+    if len(work_wires) >= 1:
+        # Lemma 7.3
+        return _decompose_mcx_with_one_worker(control_wires, target_wire, work_wires[0])
+
+    # Lemma 7.5
+    with qml.QueuingManager.stop_recording():
+        op = qml.X(target_wire)
+    return _decompose_multicontrolled_unitary(op, control_wires)
+
+
+def _decompose_multicontrolled_unitary(op, control_wires):
```

Comment:
I'm wondering if this is worth making public alongside `ctrl_decomp_zyz` and `ctrl_decomp_bisect`.  Might be useful for more than just X target decompositions.  But we can consider doing so at a later point.

Reply:
I was actually thinking the same. It is somewhat related to how we can use the same decomposition in `_decompose_multicontrolled_unitary` (although suboptimal) for Multi controlled Global Phase, and in general for any gate that `ctrl_decomp_bisect ` cannot handle.

Reply:
I can put it together in the (future) story with multi controlled Global Phase (?).